### PR TITLE
Feature/awards microservice

### PR DESCRIPTION
**Pull request by adriankulinski**

### CODE REVIEW

File: AwardsMicroservice/src/main/java/com/pgs/soft/AwardsController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class AwardsController {
+
+	@Autowired
+	AwardsService awardsService;
+
+	@Autowired
+	AwardsRepository awardsRepository;
+
+	@RequestMapping(value = "/awards")
+	public List<AwardDTO> getAllAwardsSorted(
```
Comment (kfirsztPgs) [inline AwardsMicroservice/src/main/java/com/pgs/soft/AwardsController.java]: category should not be required, and result should be groubed by category

File: AwardsMicroservice/src/main/java/com/pgs/soft/dto/AwardDTO.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+public class AwardDTO {
+
+	private String name;
+
+	private String description;
+
+	private int pointsPrice;
+
+	private String category;
```
Comment (kfirsztPgs) [inline AwardsMicroservice/src/main/java/com/pgs/soft/dto/AwardDTO.java]: Please leave enum Category here

File: AwardsMicroservice/src/test/java/com/pgs/soft/AwardsMicroserviceApplicationTests.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = AwardsMicroserviceApplication.class)
+public class AwardsMicroserviceApplicationTests {
+
+	@Test
+	public void contextLoads() {
```
Comment (kfirsztPgs) [inline AwardsMicroservice/src/test/java/com/pgs/soft/AwardsMicroserviceApplicationTests.java]: Test should not be empty

File: AwardsMicroservice/src/main/java/com/pgs/soft/AwardsController.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+			@RequestParam(value = "sortBy", defaultValue = "name", required = false) String sortProperty,
+			@RequestParam(value = "direction", defaultValue = "ASC", required = false) Sort.Direction direction,
+			@RequestParam(value = "category") String category) {
```
Comment (kfirsztPgs) [inline AwardsMicroservice/src/main/java/com/pgs/soft/AwardsController.java]: Change category param to enum

File: AwardsMicroservice/src/main/java/com/pgs/soft/AwardsController.java (hunk 1 above)
Comment (adriankulinski) [inline AwardsMicroservice/src/main/java/com/pgs/soft/AwardsController.java]: Done.

File: AwardsMicroservice/src/main/java/com/pgs/soft/dto/AwardDTO.java (hunk 2 above)
Comment (adriankulinski) [inline AwardsMicroservice/src/main/java/com/pgs/soft/dto/AwardDTO.java]: Done

File: AwardsMicroservice/src/test/java/com/pgs/soft/AwardsMicroserviceApplicationTests.java (hunk 3 above)
Comment (adriankulinski) [inline AwardsMicroservice/src/test/java/com/pgs/soft/AwardsMicroserviceApplicationTests.java]: Done

File: AwardsMicroservice/src/main/java/com/pgs/soft/AwardsController.java (hunk 4 above)
Comment (adriankulinski) [inline AwardsMicroservice/src/main/java/com/pgs/soft/AwardsController.java]: Done

File: AwardsMicroservice/src/main/java/com/pgs/soft/service/impl/AwardServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+		awardDTO.setPointsPrice(award.getPointsPrice());
+
+		return awardDTO;
+	}
+
+	@Override
+	public Map<Category, List<AwardDTO>> getAllAwardsGrouped() {
+		Map<Category, List<AwardDTO>> awardsGrouped = new HashMap<>();
+		Sort sort = new Sort("name");
+		for (Category category : Category.values()) {
```
Comment (kfirsztPgs) [inline AwardsMicroservice/src/main/java/com/pgs/soft/service/impl/AwardServiceImpl.java]: Please change it to one query and then map (what if there will be 100 categories?)

File: AwardsMicroservice/src/main/java/com/pgs/soft/service/impl/AwardServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+			awardsGrouped.put(category, getAwardsByCategoryAndSorted(category, sort));
+		}
+
+		// List<AwardDTO> list1 = getAwardsByCategoryAndSorted(category, new
```
Comment (kfirsztPgs) [inline AwardsMicroservice/src/main/java/com/pgs/soft/service/impl/AwardServiceImpl.java]: To be removed

File: AwardsMicroservice/src/main/java/com/pgs/soft/service/impl/AwardServiceImpl.java (hunk 5 above)
Comment (adriankulinski) [inline AwardsMicroservice/src/main/java/com/pgs/soft/service/impl/AwardServiceImpl.java]: Done :)

File: AwardsMicroservice/src/main/java/com/pgs/soft/service/impl/AwardServiceImpl.java (hunk 6 above)
Comment (adriankulinski) [inline AwardsMicroservice/src/main/java/com/pgs/soft/service/impl/AwardServiceImpl.java]: Done.

**Pull request merged**